tests/python/unittest/test_operator.py

```diff
@@ -2348,70 +2350,71 @@ def bilinear_forward_numpy(data, grid):
-
-                        topLeftDotProduct = 0
-                        topRightDotProduct = 0
-                        bottomLeftDotProduct = 0
-                        bottomRightDotProduct = 0
-
-                        for channel in range(num_channel):
-                            # left top
-                            if between(xInTopLeft,0,input_width-1) and between(yInTopLeft,0,input_height-1):
-                                topLeftDotProduct += data[i,channel,yInTopLeft, xInTopLeft] * \
-                                    out_grad[i,channel,yout,xout]
-                                data_grad[i, channel, yInTopLeft, xInTopLeft] += xWeightTopLeft * \
-                                    yWeightTopLeft * out_grad[i,channel,yout,xout]
-                            # right top
-                            if between(xInTopLeft+1,0,input_width-1) and between(yInTopLeft,0,input_height-1):
-                                topRightDotProduct += data[i, channel, yInTopLeft,xInTopLeft+1] * \
-                                    out_grad[i, channel, yout,xout] 
-                                data_grad[i, channel,yInTopLeft, xInTopLeft+1] += (1-xWeightTopLeft) * \
-                                    yWeightTopLeft * out_grad[i,channel,yout,xout]
-                            # left bottom
-                            if between(xInTopLeft,0,input_width-1) and between(yInTopLeft+1,0,input_height-1):
-                                bottomLeftDotProduct += data[i, channel,yInTopLeft+1, xInTopLeft] * \
-                                    out_grad[i,channel,yout,xout]
-                                data_grad[i,channel,yInTopLeft+1,xInTopLeft]+=xWeightTopLeft * \
-                                    (1-yWeightTopLeft)* out_grad[i,channel,yout,xout]
-                            # right bottom
-                            if between(xInTopLeft+1,0,input_width-1) and between(yInTopLeft+1,0,input_height-1):
-                                bottomRightDotProduct += data[i,channel,yInTopLeft+1, xInTopLeft+1] * \
-                                    out_grad[i,channel,yout,xout]
-                                data_grad[i,channel,yInTopLeft+1,xInTopLeft+1]+= (1-xWeightTopLeft) * \
-                                    (1-yWeightTopLeft)*out_grad[i,channel,yout,xout]
-
-                        yf = -xWeightTopLeft * topLeftDotProduct + xWeightTopLeft*bottomLeftDotProduct - \
-                            (1-xWeightTopLeft)* topRightDotProduct + (1-xWeightTopLeft)*bottomRightDotProduct
-                        xf = -yWeightTopLeft * topLeftDotProduct + yWeightTopLeft*topRightDotProduct - \
-                            (1-yWeightTopLeft)*bottomLeftDotProduct + (1-yWeightTopLeft)*bottomRightDotProduct
-                        grid_grad[i,0,yout,xout] = xf * (input_width-1) / 2.0
-                        grid_grad[i,1,yout,xout] = yf * (input_height-1) / 2.0
-
+                    xInTopLeft = int(floor(xcoord))
```

Comment:
remove the wrong if condition:

if xcoord>=0 and xcoord<=input_width-1 and ycoord>=0 and ycoord<=input_height-1: